dapp/client/App.elm

```diff
@@ -67,19 +73,33 @@ update : Message -> Model -> ( Model, Cmd Message )
 update message model =
     case message of
         AddressChanged newAddress ->
-            ( { model | address = Just newAddress }, Cmd.none )
+            ( { initialModel | address = Just newAddress }, Cmd.none )
```

Comment:
Whenever the address changes, go back to `initialModel` because we'll want to request the wallet again and any other values we may store in the model in the future will probably all need to be reset at this time.